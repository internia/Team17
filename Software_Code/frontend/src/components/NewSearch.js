import React, { Component } from 'react'
// import Button from 'react-bootstrap/Button'
import Row from 'react-bootstrap/Row';
// import Col from 'react-bootstrap/Col';
import styled from 'styled-components';
import { Col } from 'react-bootstrap';



const Styles = styled.div`

    --bg: #ffffffee;
    --fg: #424242;
    --bd: #BDBDBD;

    //border-radius: 14px;
    background: #ffffff40;
    width: auto;
    //margin: 50px 0px 80px 0px;
    padding 30px;

    // .row {
    //     margin: '0px 0px 10px 0px', 
    //     width: auto;
    //     background: blue;

    // }

    .smallLabel {
        color: white;
        font-weight: bold;
        font-size: small;
        padding-left: 4px;
    }

    .label {
        color: grey;
        height: auto;
        margin: 10px 0 0 0;
        width: 15%;
    }

    .radio{
        width: 3%;
        margin: 17px 0px 0px 0px;   
    }

    .mainSearch {
        width: 100%;
        height: 3em;
        background: var(--bg);
        color: var(--fg);
        border: 3px solid #0E9DD9;
        border-radius: 9px;
        border-width: 2px;
        padding: 12px;
    }  
    
    .locSearch {
        width: 100%;
        height: 3em;
        padding: 1em;
        border-width: 2px;
        border: 2px solid #0E9DD9;
        border-radius: 9px;
        background: var(--bg);
        cursor: pointer;
    }

    .radSearch {
        width: 100%;
        background: var(--bg);
        height: 3em;
        color: var(--fg);
        border: 2px solid #0E9DD9;
        border-width: 2px;
        border-radius: 9px;
        padding: 12px;
    }

    .butn {
        width: 100%;
        background: var(--bg);
        color: var(--fg);
        border: 2px solid #0E9DD9;
        padding-bottom: 10px;
        padding-top: 10px;
        margin: 0px;
        margin-top: 5px;
        border-radius: 9px;
    }  
`;

var hStyle;

export default class NewSearch extends Component {
    constructor() {
        super();
        this.state = {
            searchMain: "",
            searchLocation: "",
            searchRadius: "",
            selectedOption: "",
            searchMin: "0",
            searchMax: "99999999999999999",
        }
    }

    changeHStyle() {
        if (this.props.home === true) {
            hStyle = {
                margin: '50px 0px 80px 0px',
                width: "100%",
                borderRadius: '14px',
            }
        } else {
            hStyle = {
                margin: '0px 0px 0px 0px',
                width: "100%",
                borderRadius: '0px'
            }
        }
    }

    sendData = () => {
        this.props.parentCallback(this.state);
    }

    handleSubmit = (event) => {
        console.log('Submit handling')
        event.preventDefault();
        this.sendData()
    }

    updateMainSelectedOption() {
        if (isNaN(this.state.searchMain)) {
            this.setState({
                selectedOption: 'procName',
            });
        } else {
            this.setState({
                selectedOption: 'procCode',
            });
        }
    }

    updateMainSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ searchMain: event.target.value })
    }
    updateLocSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ searchLocation: event.target.value })
    }
    updateRadSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ searchRadius: event.target.value })
    }
    updateMaxSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ searchMax: event.target.value })
    }
    updateMinSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ searchMin: event.target.value })
    }
    updateSearch(event) {
        this.updateMainSelectedOption()
        this.setState({ search: event.target.value })
    }

    render() {
        this.changeHStyle()
        return (
            <form onSubmit={this.handleSubmit} style={{ width: "100%" }}>
                <Styles style={hStyle}>
                    {/* <label className="label">Search By: </label> */}
                    <Row>

                    </Row>
                    <Row>

                        <Col sm='8'>
                            <label className="smallLabel">Search for procedure</label>
                        </Col>

                        <Col sm='2'>
                            <label className="smallLabel">ZIP code</label>
                        </Col>

                        <Col sm='2'>
                            <label className="smallLabel">Search Radius (Miles)</label>
                        </Col>

                    </Row>

                    <Row>

                        <Col sm='4'>
                            <input className="mainSearch"
                                placeholder="Procedure name or DRG code"
                                type="text"
                                onChange={this.updateMainSearch.bind(this)} />
                        </Col>

                        <Col sm='2'>
                            <input className="locSearch"
                                placeholder="Zip code"
                                type="text"
                                onChange={this.updateLocSearch.bind(this)}
                                />
                        </Col>

                        <Col sm='2'>
                            <input className="radSearch"
                                type="text"
                                placeholder="Minimum"
                                onChange={this.updateMinSearch.bind(this)} />
                        </Col>
                        <Col sm='2'>
                            <input className="radSearch"
                                type="text"
                                placeholder="Maximum"
                                onChange={this.updateMaxSearch.bind(this)} />
                        </Col>

                        <Col sm='2'>
                            <input className="radSearch"
                                type="text"
                                placeholder="Distance"
                                onChange={this.updateRadSearch.bind(this)} />
                        </Col>

                    </Row>
                    <Row className={'align-items-right'}>
                        <Col className={'offset-8'} sm='4'>
                            <input className={'butn'}
                                type="submit"
                                value="Submit"
                                onClick={this.handleSubmit} />
                        </Col>
                    </Row>
                </Styles>
            </ form>
        )
    }
}

// {
//     <Col>
//         <input className="subSearch"
//             type="text"
//             value={this.state.search}
//             onChange={this.updateSearch.bind(this)} />
//     </Col>
//         <Col>
//             <input className="subSearch"
//                 type="text"
//                 value={this.state.search}
//                 onChange={this.updateSearch.bind(this)} />
//         </Col> 
// }