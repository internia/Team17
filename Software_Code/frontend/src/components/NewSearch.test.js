import React from 'react';
import { render } from '@testing-library/react';
import NewSearch from './NewSearch';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { isCompositeComponent } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() })


it('renders without crashing', ()=>{
  const div = document.createElement("div");
  ReactDOM.render(<NewSearch></NewSearch>, div)
})

it('renders searchbar correctly', () => {
  function callBack(){
    console.log("called back")
  }
  render(<NewSearch home={false} 
    parentcallback={callBack} ></NewSearch>)
})


it("matches snapshot", ()=> {
  function callBack(){
    console.log("called back")
  }
  const tree =  renderer.create(<NewSearch home={false} 
  parentcallback={callBack}></NewSearch>)
  expect(tree).toMatchSnapshot();
})