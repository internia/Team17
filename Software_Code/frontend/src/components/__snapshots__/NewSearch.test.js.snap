// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="sc-AykKC bkGdwC"
    style={
      Object {
        "borderRadius": "0px",
        "margin": "0px 0px 0px 0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
    />
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <label
          className="smallLabel"
        >
          Search for procedure
        </label>
      </div>
      <div
        className="col-sm-4"
      >
        <label
          className="smallLabel"
        >
          State
        </label>
      </div>
      <div
        className="col-sm-2"
      >
        <label
          className="smallLabel"
        >
          Search Radius (KM)
        </label>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-6"
      >
        <input
          className="mainSearch"
          onChange={[Function]}
          placeholder="Procedure name or DRG code"
          type="text"
        />
      </div>
      <div
        className="p-0 col-sm-4"
      >
        <div
          class="form-group m-0"
        >
          <div
            class="col-sm-12"
            style={
              Object {
                "height": "3em",
              }
            }
          >
            <select
              class="form-control locSearch"
              id="state"
              name="state"
              onChange={[Function]}
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <option
                value=""
              >
                N/A
              </option>
              <option
                value="AK"
              >
                Alaska
              </option>
              <option
                value="AL"
              >
                Alabama
              </option>
              <option
                value="AR"
              >
                Arkansas
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="CA"
              >
                California
              </option>
              <option
                value="CO"
              >
                Colorado
              </option>
              <option
                value="CT"
              >
                Connecticut
              </option>
              <option
                value="DC"
              >
                District of Columbia
              </option>
              <option
                value="DE"
              >
                Delaware
              </option>
              <option
                value="FL"
              >
                Florida
              </option>
              <option
                value="GA"
              >
                Georgia
              </option>
              <option
                value="HI"
              >
                Hawaii
              </option>
              <option
                value="IA"
              >
                Iowa
              </option>
              <option
                value="ID"
              >
                Idaho
              </option>
              <option
                value="IL"
              >
                Illinois
              </option>
              <option
                value="IN"
              >
                Indiana
              </option>
              <option
                value="KS"
              >
                Kansas
              </option>
              <option
                value="KY"
              >
                Kentucky
              </option>
              <option
                value="LA"
              >
                Louisiana
              </option>
              <option
                value="MA"
              >
                Massachusetts
              </option>
              <option
                value="MD"
              >
                Maryland
              </option>
              <option
                value="ME"
              >
                Maine
              </option>
              <option
                value="MI"
              >
                Michigan
              </option>
              <option
                value="MN"
              >
                Minnesota
              </option>
              <option
                value="MO"
              >
                Missouri
              </option>
              <option
                value="MS"
              >
                Mississippi
              </option>
              <option
                value="MT"
              >
                Montana
              </option>
              <option
                value="NC"
              >
                North Carolina
              </option>
              <option
                value="ND"
              >
                North Dakota
              </option>
              <option
                value="NE"
              >
                Nebraska
              </option>
              <option
                value="NH"
              >
                New Hampshire
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NM"
              >
                New Mexico
              </option>
              <option
                value="NV"
              >
                Nevada
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="OH"
              >
                Ohio
              </option>
              <option
                value="OK"
              >
                Oklahoma
              </option>
              <option
                value="OR"
              >
                Oregon
              </option>
              <option
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                value="PR"
              >
                Puerto Rico
              </option>
              <option
                value="RI"
              >
                Rhode Island
              </option>
              <option
                value="SC"
              >
                South Carolina
              </option>
              <option
                value="SD"
              >
                South Dakota
              </option>
              <option
                value="TN"
              >
                Tennessee
              </option>
              <option
                value="TX"
              >
                Texas
              </option>
              <option
                value="UT"
              >
                Utah
              </option>
              <option
                value="VA"
              >
                Virginia
              </option>
              <option
                value="VT"
              >
                Vermont
              </option>
              <option
                value="WA"
              >
                Washington
              </option>
              <option
                value="WI"
              >
                Wisconsin
              </option>
              <option
                value="WV"
              >
                West Virginia
              </option>
              <option
                value="WY"
              >
                Wyoming
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="col-sm-2"
      >
        <input
          className="radSearch"
          onChange={[Function]}
          placeholder="Distance"
          type="text"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="text-right col-sm-12"
      >
        <label />
        <input
          className="butn"
          onClick={[Function]}
          type="submit"
          value="Submit"
        />
      </div>
    </div>
  </div>
</form>
`;
